### PR TITLE
Update Chest API for new merging logic.

### PR DESCRIPTION
**Pull request by TheGlitch76**
[SpongeAPI](https://github.com/SpongePowered/SpongeAPI/pull/2106) | **SpongeCommon**

Chests now have much better logic for finding its connected chest. This PR updates the mixin's logic to match Vanilla's.

Also, LockableLootTileEntity's customName property was moved to its superclass in Vanilla. This PR also updates the mixins to reflect that change.

### CODE REVIEW

File: src/mixins/java/org/spongepowered/common/mixin/api/mcp/tileentity/LockableTileEntityMixin_API.java [hunk 1]
```diff
@@ -48,11 +50,13 @@
 
     @Shadow @Nullable private LockCode code;
 
+    @Nullable @Shadow private ITextComponent customName; // Not really nullable, but it should be... - Roquette
```
Comment (i509VCB) [inline src/mixins/java/org/spongepowered/common/mixin/api/mcp/tileentity/LockableTileEntityMixin_API.java:52]: Please remove the space between shadowed fields

File: src/mixins/java/org/spongepowered/common/mixin/api/mcp/tileentity/LockableLootTileEntityMixin_API.java [hunk 2]
```diff
@@ -25,27 +25,11 @@
...
-    @Nullable @Shadow protected String customName; // Not really nullable, but it should be...
-
-    @Override
-    public DataContainer toContainer() {
-        DataContainer container = super.toContainer();
-        if (this.customName != null) {
-            container.set(Constants.TileEntity.LOCKABLE_CONTAINER_CUSTOM_NAME, this.customName);
-        }
-        return container;
-    }
```
Comment (i509VCB) [inline src/mixins/java/org/spongepowered/common/mixin/api/mcp/tileentity/LockableLootTileEntityMixin_API.java:49]: Unsure about this one myself?

@gabizou @Zidane

Comment (i509VCB) [reply src/mixins/java/org/spongepowered/common/mixin/api/mcp/tileentity/LockableLootTileEntityMixin_API.java:49]: Mind the horrible code selection zone lol

Comment (TheGlitch76) [reply src/mixins/java/org/spongepowered/common/mixin/api/mcp/tileentity/LockableLootTileEntityMixin_API.java:49]: the `customName` string doesn't exist here any more--it was moved in vanilla.

**Pull request merged**